### PR TITLE
mesa: new formula, v18.2.3

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

### CODE REVIEW

**Comment by yurikoles** [comment]
There is a test app that is shipped with mesa: glxgears, but it depends on X server, I can successfully launch it from XQaurtz terminal and see that gears is rolling.

**Comment by yurikoles** [comment]
@ilovezfs could you look into this?

**Comment by yurikoles** [comment]
@BrewTestBot it doesn't build with system `expat`.

File: Formula/mesa.rb [hunk 1]
```diff
@@ -0,0 +1,368 @@
+class Mesa < Formula
+  include Language::Python::Virtualenv
+  desc "Graphics Library"
+  homepage "https://www.mesa3d.org/"
+  url "https://mesa.freedesktop.org/archive/mesa-18.2.3.tar.xz"
+  sha256 "e2bf83c17e1abdecb1ee81af22652e27e9aa38f963e95e60f34275cc0376304f"
+  depends_on "meson-internal" => :build
```
Comment (fxcoudert) [inline Formula/mesa.rb]: Blank line above

File: Formula/mesa.rb [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@2" => :build
+  depends_on "expat"
+  depends_on "freeglut" # for test
```
Comment (fxcoudert) [inline Formula/mesa.rb]: `=> :test`

File: Formula/mesa.rb [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+  depends_on :x11
+
+  resource "Mako" do
+    url "https://files.pythonhosted.org/packages/eb/f3/67579bb486517c0d49547f9697e36582cd19dafb5df9e687ed8e22de57fa/Mako-1.0.7.tar.gz"
+    sha256 "4e02fde57bd4abb5ec400181e4c314f56ac3e49ba4fb8b0d50bba18cb27d25ae"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
```
Comment (fxcoudert) [inline Formula/mesa.rb]: Is mako used only for the build, or is it also needed after the install is done?

File: Formula/mesa.rb [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+    end
+    mkdir "build" do
+      system "meson", "--prefix=#{prefix}", "-D buildtype=plain", "-D b_ndebug=true", ".."
+      system "ninja"
+      system "ninja", "install"
+    end
+  end
+
+  test do
+    (testpath/"gears.c").write <<~EOS
```
Comment (fxcoudert) [inline Formula/mesa.rb]: Use a resource for `gears.c`, or install it from the source if present (into `pkgshare`).

File: Formula/mesa.rb [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+        glutReshapeFunc(reshape);
+        glutKeyboardFunc(key);
+        glutSpecialFunc(special);
+        glutVisibilityFunc(visible);
+
+        glutMainLoop();
+        return 0;             /* ANSI C requires main to return int. */
+      }
+    EOS
+    system "cc", "#{testpath}/gears.c", "-o gears", "-L#{lib}", "-L#{HOMEBREW_PREFIX}/lib/", "-lglut", "-lgl"
```
Comment (fxcoudert) [inline Formula/mesa.rb]: Use `Formula["freeglut"].opt_lib` instead of `HOMEBREW_PREFIX`

Comment (fxcoudert) [inline Formula/mesa.rb]: `system ENV.cc, "gears.c", "-o", "gears", `…

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/mesa.rb (hunk 3 above)
Comment (yurikoles) [reply Formula/mesa.rb]: According to [ArchLinux package](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/mesa#n10)

File: Formula/mesa.rb (hunk 5 above)
Comment (yurikoles) [reply Formula/mesa.rb]: I'm not sure how to correctly use double quotes inside a double quotes?

File: Formula/mesa.rb (hunk 4 above)
Comment (yurikoles) [reply Formula/mesa.rb]: How to correctly stage this kind of resource to reuse it in test?

File: Formula/mesa.rb (hunk 3 above)
Comment (fxcoudert) [reply Formula/mesa.rb]: Then you need to install mako inside `buildpath`, not `libexec`.

Comment (yurikoles) [reply Formula/mesa.rb]: Done

**Comment by yurikoles** [comment]
@fxcoudert bump?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @yurikoles for the pull request

**Pull request closed without merging**